### PR TITLE
chore: bump scala.js to 1.1.0

### PR DESCRIPTION
**Pull request by ngbinh**
in preparation for scala.js 1.X for `endpoints`

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -62,17 +60,17 @@ val `faithful-cats` =
     .settings(
       crossScalaVersions := crossScalaV,
       libraryDependencies ++= Seq(
-        "org.typelevel" %%% "cats-core" % "2.0.0",
-        "org.typelevel" %%% "cats-laws" % "2.0.0" % Test,
-        "org.typelevel" %%% "discipline-scalatest" % "1.0.0-M1" % Test,
-        "org.scalatest" %%% "scalatest" % "3.1.0-M2" % Test
+        "org.typelevel" %%% "cats-core" % "2.2.0-M2",
```
Comment (julienrf) [inline build.sbt]: Is there a non-milestone release that supports Scala.js 1.x?

Comment (ngbinh) [reply build.sbt]: yes, use `2.1.1` the the new commit

File: build.sbt [hunk 2]
```diff
@@ -62,17 +60,17 @@ val `faithful-cats` =
...
       crossScalaVersions := crossScalaV,
       libraryDependencies ++= Seq(
-        "org.typelevel" %%% "cats-core" % "2.0.0",
-        "org.typelevel" %%% "cats-laws" % "2.0.0" % Test,
-        "org.typelevel" %%% "discipline-scalatest" % "1.0.0-M1" % Test,
-        "org.scalatest" %%% "scalatest" % "3.1.0-M2" % Test
+        "org.typelevel" %%% "cats-core" % "2.1.1",
+        "org.typelevel" %%% "cats-laws" % "2.1.1" % Test,
+        "org.typelevel" %%% "discipline-scalatest" % "1.0.1" % Test,
+        "org.scalatest" %%% "scalatest" % "3.2.0-M4" % Test
```
Comment (julienrf) [inline build.sbt]: Same here: what do you think about using 3.1.x?

Comment (ngbinh) [reply build.sbt]: Yes, down to `3.1.2` now

File: build.sbt [hunk 3]
```diff
@@ -1,7 +1,7 @@
 organization in ThisBuild := "org.julienrf"
 
-scalaVersion in ThisBuild := "2.13.0"
-val crossScalaV = Seq( "2.13.0", "2.12.10", "2.11.12")
+scalaVersion in ThisBuild := "2.13.2"
+val crossScalaV = Seq( "2.13.2", "2.12.12", "2.11.12")
```
Comment (julienrf) [inline build.sbt]:
```suggestion
val crossScalaV = Seq( "2.13.2", "2.12.11")
```

It seems that cats has dropped support for 2.11?

**Review by julienrf — COMMENTED** [review]
Thanks @ngbinh!

I have the following error when I try to compile the project:

~~~
[error] /home/julien/workspace/dev/julienrf/faithful/faithful-cats/src/test/scala/faithful/cats/TestSuite.scala:18:42: illegal inheritance;
[error]  self-type faithful.cats.TestSuite does not conform to org.typelevel.discipline.scalatest.Discipline's selftype org.typelevel.discipline.scalatest.Discipline with org.scalatest.prop.Configuration
[error] class TestSuite extends AnyFunSuite with Discipline {
[error]                                          ^
[error] /home/julien/workspace/dev/julienrf/faithful/faithful-cats/src/test/scala/faithful/cats/TestSuite.scala:34:11: could not find implicit value for parameter config: TestSuite.this.PropertyCheckConfiguration
[error]   checkAll("Future[Int]", MonadErrorTests[Future, Throwable].monadError[Int, Int, Int])
[error]           ^
~~~

File: build.sbt (hunk 3 above)
Comment (ngbinh) [reply build.sbt]: yes, let's drop it

**Comment by ngbinh** [comment]
@julienrf the tests are compiled and run now. Also drop scala 2.11 as suggested

**Comment by julienrf** [comment]
Thank you!

**Comment by julienrf** [comment]
`org/julienrf/faithful_sjs1_2.13/2.0.0/faithful_sjs1_2.13-2.0.0.jar` is on its way to maven central

**Comment by ngbinh** [comment]
great, thanks

**Pull request merged**